arviz/data/io_cmdstan.py

```diff
@@ -430,13 +436,23 @@ def log_likelihood_to_xarray(self):
             data = _unpack_ndarrays(chain_data, columns, self.dtypes)
             data_warmup = _unpack_ndarrays(chain_data_warmup, columns, self.dtypes)
         else:
-            if isinstance(log_likelihood, str):
-                log_likelihood = [log_likelihood]
-            columns = {
-                col: idx
-                for col, idx in self.posterior_columns.items()
-                if any(item == col.split(".")[0] for item in log_likelihood)
-            }
+            if isinstance(log_likelihood, dict):
+                log_lik_to_obs_name = {v: f"{k}.{{}}".format for k, v in log_likelihood.items()}
+                columns = {
+                    log_lik_to_obs_name[col](varidx): idx
+                    for col, varidx, idx in (
+                        (*col.split("."), idx) for col, idx in self.posterior_columns.items()
```

Comment:
Is there a possibility for something like `log_lik.1.2.3`?

Reply:
Thanks @ahartikainen , good catch! I guess that's a possibility. I've just committed a tweak to split only on one dot into two strings: column name and indices. I am not sure though how that is handled down the line, does it look correct to you?

Reply:
It looks good.